### ISSUE TITLE
Support for functools.total_ordering decorator

### ISSUE THREAD

**Issue by lovetox** [microsoft/pyright#2766]
**Support for functools.totalordering decorator**

```python

from __future__ import annotations

from functools import total_ordering
from enum import Enum

ORDER = {
    'owner': 0,
    'admin': 1}

@total_ordering
class Affiliation(Enum):
    OWNER = 'owner'
    ADMIN = 'admin'

    def __gt__(self, other: Affiliation) -> bool:
        return ORDER[self.value] > ORDER[other.value]

a = Affiliation('admin')
b = Affiliation('owner')
fails = a < b
works = a > b
```
This currently raises `operator < not supported - GeneralTypeIssue`

**Describe the solution you'd like**
when decorator is discovered show all comparisons

`__lt__(), __le__(), __gt__(), or __ge__()`

as supported

**Additional context**
[functools documentation](https://docs.python.org/3/library/functools.html#functools.total_ordering)

**Comment by erictraut** [comment]
This requires some rather ugly special-casing in the type checker for this specific decorator. However, it is a stdlib function, so it might be justified. Mypy has already included special-case support for this function, so that's good signal that it's popular enough to justify adding the support.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.201, which I just published. It will also be included in the next release of pylance.

**Comment by lovetox** [comment]
great thanks !
